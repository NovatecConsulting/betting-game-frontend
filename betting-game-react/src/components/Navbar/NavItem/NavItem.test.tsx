import React from "react";
import { shallow } from "enzyme";
import { NavLink } from "react-router-dom";
import { NavItem } from ".";

describe(NavItem.name + " component", () => {
    it("should render and match snapshot with correct content.", () => {
        const wrapper = shallow(
            <NavItem
                buttonURL={"/testButtonURL"}
                buttonText={"TestButtonText"}
            />
        );

        expect(wrapper.find(NavLink).prop("to")).toEqual("/testButtonURL");
        expect(wrapper.find(NavLink)).toIncludeText("TestButtonText");
        expect(wrapper).toMatchSnapshot();
    });
});
