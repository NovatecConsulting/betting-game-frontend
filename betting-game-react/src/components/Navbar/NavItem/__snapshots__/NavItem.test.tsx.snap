// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavItem component should render and match snapshot with correct content. 1`] = `
<NavLink
  activeClassName="border-gray-400 text-indigo-100"
  className="mr-4 lg:inline-block lg:mt-0 py-2 border-b-2 border-transparent inline-block text-gray-400 hover:text-white"
  to="/testButtonURL"
>
  <span>
    TestButtonText
  </span>
</NavLink>
`;
